werkit/lambda/parallel.py

```diff
@@ -0,0 +1,28 @@
+import boto3
+import json
+import asyncio
+import os
+
+client = boto3.client('lambda')
+
+LAMBDA_WORKER_FUNCTION_NAME = 'LAMBDA_WORKER_FUNCTION_NAME'
+
+if LAMBDA_WORKER_FUNCTION_NAME not in os.environ:
+    raise Error(f'Environment variable {LAMBDA_WORKER_FUNCTION_NAME} must be defined to use s3_upload_handler')
+
+lambda_worker_function_name = os.environ[LAMBDA_WORKER_FUNCTION_NAME]
+
+#TODO: how to handle errors
+async def call_worker_service(y):
+    response = client.invoke(
+        FunctionName=lambda_worker_function_name,
+        Payload=json.dumps(y)
+    )
+    return json.load(response['Payload'])
+
+async def synchronous_map_on_lambda(x):
```

Comment:
What is synchronous about this? Ahh, it's that it blocks until it's done. I think that's confusing. Synchronous makes it sound like it does one at a time. Maybe `parallel_map_on_lambda` would be clearer.

I'd probably call this parameter `obj` if it's expected to be a list or a dict.

It occurs to me though that this might not make sense. When we're building an object of results, we are starting with an array of keys and then want to return an object. Maybe we should have a flag `ret_dict=False` which can be set, and determines whether a list is returned, or an object is returned, with the values in the input list used as keys.

Reply:
> What is synchronous about this? Ahh, it's that it blocks until it's done. I think that's confusing. Synchronous makes it sound like it does one at a time. Maybe parallel_map_on_lambda would be clearer.

Sounds good

Reply:
> I'd probably call this parameter obj if it's expected to be a list or a dict.

> It occurs to me though that this might not make sense. When we're building an object of results, we are starting with an array of keys and then want to return an object. Maybe we should have a flag ret_dict=False which can be set, and determines whether a list is returned, or an object is returned, with the values in the input list used as keys.

I can think of 3 possible options for this parameter:

1. We expect a dictionary output, and a dictionary input of the form `{ key -> arguments }`
2. We expect a dictionary output, and a list of pairs as input of the form `[ (key, (arguments), ...) ]`
3. We expect a list output, and a list of anything as input: `[ anything, ... ]`

The nice thing about supporting option 2. is that it is ordered, and easy to prepare through list comprehension syntax. You can also pass it to a dict or OrderedDict constructor. 
We might want to support 1 as a convenience API.

We would not need 3. in our application (because we would be returning a list, rather than a dict), but might want to support it if we anticipate it being useful for other users/applications.